apps/bgu/bgu_generator.cpp

```diff
@@ -126,6 +127,86 @@ Matrix<M, N> solve(Matrix<M, M> A, Matrix<M, N> b, Func compute, Var at, bool sk
     return b;
 };
 
+// Solve Ax = b at each x, y, z exploiting the fact that A is
+// symmetric. Compute the result at the given Func and Var.
+template<int M, int N>
```

Comment:
Maybe give a reference to how square root free Cholesky works: https://users.wpi.edu/~walker/MA514/HANDOUTS/cholesky.pdf ?

Reply:
Oh wow, I would have loved that reference when I was writing this. I just made it up as I went along by messing with regular Cholesky.